Sources/OSLogClient/Public/LogClient.swift

```diff
@@ -0,0 +1,229 @@
+        }
+        immediatePollTaskMap[taskId] = pollTask
+    }
+    
+    /// Will assign the given Bool flag to the ``LogClient/shouldPauseIfNoRegisteredDrivers`` property.
+    ///
+    /// - Note: When `true` if all drivers are deregistered, polling will cease until valid drivers are registered again.
+    /// - Parameter flag: Bool whether enabled.
+    public func setShouldPauseIfNoRegisteredDrivers(_ flag: Bool) async {
+        await config.setShouldPauseIfNoRegisteredDrivers(flag)
+    }
+
+    // MARK: - Helpers: Polling
+    
+    /// Will cancel any pending poll tasks, but not assign the `isPollingEnabled` flag to false.
+    /// This is called when all drivers are deregistered but the consumer has not explicitly stopped polling.
+    func softStopPolling() {
+        pendingPollTask?.cancel()
+        pendingPollTask = nil
+    }
+
+    /// Will assess the given entry and ensure it is both a valid `OSLogEntryLog` instance.
+    /// - Parameter entry: The entry to assess.
+    /// - Returns: `OSLogEntryLog` or `nil`
+    func validateLogEntry(_ entry: OSLogEntry) -> OSLogEntryLog? {
+        entry as? OSLogEntryLog
+    }
+
+    /// Will poll logs since the last processed time position and send to any registered drivers for validation and processing.
+    /// - Parameter date: Optional date to request logs from. Leave this `nil` to default to the `lastProcessed` property.
+    func pollLatestLogs(from date: Date? = nil) async {
+        do {
+            var predicate: NSPredicate?
+            let lastProcessed = await config.lastProcessed
+            if let fromDate = date ?? lastProcessed {
+                predicate = datePredicate.withSubstitutionVariables(["DATE": fromDate])
+            }
+            let items = try logStore.getEntries(matching: predicate).compactMap(validateLogEntry)
+            let sortedItems = items.sorted(by: { $0.date <= $1.date })
+            // Broadcast logs to drivers
+            await sortedItems.asyncForEach { log in
+                await logDriverRegistry.drivers.asyncForEach {
+                    $0.processLogIfValid(log)
+                }
+            }
+            let nextLastProcessed = sortedItems.last?.date ?? lastProcessed
+            await config.setLastProcessed(nextLastProcessed)
+        } catch {
+            logger.error("Error: Unable to get log entries from store: `\(error.localizedDescription)`")
+        }
+    }
+
+    // MARK: - Helpers: Internal
+
+    /// Will execute the poll operation on the log poller instance.
+    func executePoll() async {
+        guard await isPollingEnabled else {
+            pendingPollTask = nil
+            return
+        }
+        pendingPollTask = Task(priority: .userInitiated) { [weak self] in
+            guard let self else { return }
+            try await Task.sleep(nanoseconds: self.pollingInterval.nanoseconds)
+            await self.pollLatestLogs()
+            await self.executePoll()
+        }
```

Comment:
ahh yep good call - will add

Reply:
by `before invoking recursively` meaning before the `executePoll` is invoked again? i.e

```swift
pendingPollTask = Task(priority: .userInitiated) { [weak self] in
    guard let self else { return }
    try await Task.sleep(nanoseconds: self.pollingInterval.nanoseconds)
    try Task.checkCancellation()
    await self.pollLatestLogs()
    try Task.checkCancellation()
    await self.executePoll()
}
```

Reply:
Yep, not sure if overkill. But given I've seen polling take quite some time, like longer than the poll interval, it seems it might be valuable. But as I mentioned in the comment, maybe it's not a problem since it's just a recursive call which _should_ exit without kicking off a new task.

Reply:
latest commits have these changes